src/quark-core/src/QuarkWallet.sol

```diff
@@ -249,7 +254,9 @@ contract QuarkWallet is IERC1271 {
 
         emit ExecuteQuarkScript(msg.sender, scriptAddress, nonce, ExecutionType.Direct);
 
-        return stateManager.setActiveNonceAndCallback(nonce, scriptAddress, scriptCalldata);
+        stateManager.setNonce(nonce);
```

Comment:
We should rename `setNonce`, I think? `set` and `clear` are a bit weird? `claimNonce` or something might be a bit clearer that it's a "take it or revert"-style function.